lib/appium/AppiumRunner.js

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env node
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/* jshint node: true */
+
+"use strict";
+
+var fs               = require("fs");
+var path             = require("path");
+var util             = require("./helpers/util");
+var logger           = require("../utils").logger;
+var MedicReporter    = require("./helpers/MedicReporter");
+var wd               = require("wd");
+var wdHelper         = require("./helpers/wdHelper");
+var screenshotHelper = require("./helpers/screenshotHelper");
+var kill             = require("tree-kill");
+var child_process    = require("child_process");
+var expectTelnet     = require("expect-telnet");
+var shell            = require("shelljs");
+var Jasmine          = require("jasmine");
+var unorm            = require("unorm");
+var elementTree      = require("elementtree");
+var q                = require("q");
+var jasmineReporters = require("jasmine-reporters");
+
+var KILL_SIGNAL = "SIGINT";
+var SMALL_BUFFER_SIZE = 1024 * 1024;
+var BIG_BUFFER_SIZE = 50 * 1024 * 1024;
+var APPIUM_SERVER_PATH = getAppiumServerPath();
+
+function AppiumRunner(options) {
+    this.options = options;
+    this.findTests();
+    this.setGlobals();
+}
+
+function getAppiumServerPath() {
+    return path.resolve(process.cwd(), "cordova-paramedic/node_modules/appium/build/lib/main.js");
+}
+
+function getFullAppPath(appPath) {
+    var fullPath = appPath;
+    if (!path.isAbsolute(appPath)) {
+        fullPath = path.join(__dirname, "../..", appPath);
+    }
+    return fullPath;
+}
+
+function getPackagePath(options) {
+    if (options.sauce) {
+        return options.sauceAppPath;
+    }
+
+    var fullAppPath = getFullAppPath(options.appPath);
+
+    switch (options.platform) {
+    case "android":
+        return path.join(fullAppPath, "/platforms/android/build/outputs/apk/android-debug.apk");
```

Comment:
What if `android-debug.apk` does not exist? You'd probably need to check for the existence and throw the error here